crates/nu-protocol/src/engine/engine_state.rs

```diff
@@ -833,6 +833,11 @@ impl EngineState {
             .expect("internal error: missing block")
     }
 
+    /// Optionally get a block by id, if it exists
+    pub fn try_get_block(&self, block_id: BlockId) -> Option<&Block> {
```

Comment:
Oh, sorry, I had a branch that also added closure support, and this was necessary because we don't want a plugin passing an invalid block ID to crash Nushell. But it's not needed for this, so it can be removed.